js/screens/SignUp/SignUpScreen.js

```diff
@@ -39,43 +39,37 @@ class SignUpScreen extends Component {
     this.props.dispatch(fetchPassword(text));
   };
 
-  userError = error => {
-    this.props.dispatch(newUserError(error));
+  _signInAsync = async ({ newUser }) => {
+    await AsyncStorage.setItem('userToken', {
+      newUser
+    });
+    this.props.navigation.navigate('Profile');
   };
 
-  addUser() {
+  addUser = () => {
+    const { email, password, firstName, lastName, users } = this.props;
+    const signInSuccess = isValidEmailAndPassword(email, password);
+
     const errorMessage = {
-      code: 'Invalid email or password',
-      message:
-        'Please Enter a valid email and password of at least 6 characters'
+      code: 'Invalid login',
+      message: 'Please Enter a Valid Email Address'
     };
+
     const userMessage = {
-      code: 'User already exists',
-      message: 'It looks like you already have an account!'
+      code: 'User Already Exists',
+      message: 'You Already Have an Account!'
```

Comment:
You do not need to create an object if you want to dispatch an error. If you want to get an error message from your firebase, all you have to do it error.message in your catch. 

If you want to dispatch custom errors, you can just pass a string of error instead of creating an object. 

Please make the requested changes.